src/main.cpp

```diff
@@ -57,8 +54,8 @@ int main(int argc, char** argv) {
       cxxopts::value<bool>(cl_args.geometry)->default_value("false"))
     ("i,input-file", "Read input from 'input-file' rather than from stdin",
       cxxopts::value<std::string>(cl_args.input_file))
-    ("l,limit", "Stop solving process after 'limit' seconds",
-      cxxopts::value<std::string>(limit_arg)->default_value(""))
+    ("l,limit", "Stop solving process after 'limit' seconds. Default 0, i.e. unlimited.",
+      cxxopts::value<unsigned>(limit_arg)->default_value("0"))
```

Comment:
This won't work since 0 is a valid value for `-l` which does not mean "no limit" but result in 0 time left for local search. You can probably keep the previous logic of using a string (default empty) and calling `std::stof` with a try block if the string is not empty.

Reply:
ah pity:) if `-1` (and checking < 0) is not an option for you I'll use the string strategy again

Reply:
argh unsigned haha I'll go for string

Reply:
can't say it got prettier with nested try/catch but found it still to be more elegant solution. ok for you?